engine/src/main/java/org/terasology/engine/rendering/nui/internal/NUIManagerInternal.java

```diff
@@ -365,7 +365,8 @@ public <T extends CoreScreenLayer> T createScreen(ResourceUrn screenUri, Class<T
                 }
                 return screen;
             } else {
-                logger.error("Screen '{}' is a '{}' and not a '{}'", screenUri, root.getClass(), expectedType);
+                logger.atError().addArgument(screenUri).addArgument(() -> root.getClass()).addArgument(expectedType).
+                        log("Screen '{}' is a '{}' and not a '{}'");
```

Comment:
The `getClass()` call should be pretty cheap. also, it's only done in the rare case that the screen type doesn't match.
```suggestion
                logger.error("Screen '{}' is a '{}' and not a '{}'", screenUri, root.getClass(), expectedType); //NOPMD
```